src/scenes/ProgressReports/EditForm/Steps/ProgressStep/ProgressStep.tsx

```diff
@@ -0,0 +1,239 @@
+import { useMutation } from '@apollo/client';
+import {
+  Box,
+  Grid,
+  ToggleButton,
+  ToggleButtonGroup,
+  Typography,
+} from '@mui/material';
+import { groupBy } from 'lodash';
+import { useState } from 'react';
+import { useDialog } from '~/components/Dialog';
+import {
+  StepProgressFragment,
+  UpdateStepProgressDocument,
+} from '../../../../Products/Detail/Progress/ProductProgress.graphql';
+import { ProgressSummaryCard } from '../../../../ProgressReports/Detail/ProgressSummaryCard';
+import { UpdatePeriodicReportDialog } from '../../../../Projects/Reports/UpdatePeriodicReportDialog';
+import { ProductTable } from '../../../Detail/ProductTable';
+import { ProgressReportCard } from '../../../Detail/ProgressReportCard';
+import { colorPalette } from '../../colorPalette';
+import { useProgressReportContext } from '../../ProgressReportContext';
+import {
+  ProgressReportEditFragment,
+  ProgressReportProgressFragment,
+} from '../../ProgressReportEdit.graphql';
+import { RoleIcon } from '../../RoleIcon';
+
+const transparentBgSx = {
+  backgroundColor: 'transparent',
+};
+
+const iconSx = {
+  height: 36,
+  width: 36,
+  mr: 0,
+};
+
+const ToggleButtonSx = (role: string) => ({
+  p: 0.25,
+  pr: 1,
+  mr: 0,
+  borderRadius: 0.6,
+  '&.Mui-selected': {
+    backgroundColor: colorPalette.stepperCard.iconBackground[role],
+  },
+  '&.Mui-selected:hover': {
+    backgroundColor: colorPalette.stepperCard.iconBackground[role],
+  },
+});
+
+const variants = ['partner', 'official'] as const;
+
+const sortCategories = (a: string, b: string) => {
+  if (a === 'Scripture') {
+    return -1;
+  }
+  if (b === 'Scripture') {
+    return 1;
+  }
+  return a.localeCompare(b);
+};
+
+export const ProgressStep = () => {
+  const { report } = useProgressReportContext();
+  const [tab, setTab] = useState<typeof variants[number]>('partner');
+
+  const [update] = useMutation(UpdateStepProgressDocument);
+
+  // Single file for new version, empty array for received date update.
+  const [dialogState, setUploading, upload] = useDialog<File[]>();
+
+  const groups: Record<
+    string,
+    Record<string, ProgressReportProgressFragment[]>
+  > = {};
+
+  report.progressForAllVariants.map((progress) => {
+    const grouped = groupBy(progress, (product) => product.product.category);
+    groups[progress[0]!.variant.key] = grouped;
+    return progress;
+  });
```

Comment:
Not using the result of `.map` is a code smell. Here's a way to do it without needing to mutate the `groups` variable.
```suggestion
  const progressByVariant = Object.fromEntries(
    report.progressForAllVariants.map((progress) => {
      const variant = progress[0]!.variant.key;
      const progressByCategory = groupBy(progress, (product) => product.product.category);
      return [variant, progressByCategory];
    })
  );
```

FWIW (in regard to above comment)
```ts
const variants = Object.keys(progressByVariant);
```